regparser/web/jobs/views.py

```diff
@@ -0,0 +1,150 @@
+from regparser.web.jobs.models import ParsingJob
+from regparser.web.jobs.serializers import ParsingJobSerializer
+from regparser.web.jobs.utils import (
+    add_redis_data_to_job_data,
+    delete_eregs_job,
+    queue_eregs_job,
+    status_url
+)
+from rest_framework.renderers import BrowsableAPIRenderer, JSONRenderer
+from rest_framework.response import Response
+from rest_framework import generics
+from rest_framework import mixins
+from rest_framework import status
+
+renderer_classes = (
+    JSONRenderer,
+    BrowsableAPIRenderer
+)
+
+
+class JobViewList(mixins.ListModelMixin,
+                  mixins.CreateModelMixin,
+                  generics.GenericAPIView):
+    queryset = ParsingJob.objects.all()
+    renderer_classes = renderer_classes
+    serializer_class = ParsingJobSerializer
+
+    def filter_queryset(self, request, *args, **kwargs):
+        """
+        Overridden in order to get data from the Redis queue as well as the DB.
+
+        Impure
+            Pulls information from the DB and the Redis queue.
+
+        :arg HttpRequest request: the incoming request.
+
+        :rtype: list[ParsingJob]
+        :returns: List of ParsingJob objects.
+        """
+        queryset = super(JobViewList, self).filter_queryset(request, *args,
+                                                            **kwargs)
+        queryset = add_redis_data_to_job_data(queryset)
+        return queryset
+
+    def create(self, request, *args, **kwargs):
+        """
+        Overridden in order to add the new job to the Redis queue.
+
+        Side effects
+            Via ``queue_eregs_job`` and ``ParsingJobSerializer.save``, alters
+            the redis queue and the DB.
+
+        :arg HttpRequest request: the incoming request.
+
+        :rtype: Response
+        :returns: JSON or HTML of the information about the job (status 201),
+            or about why the job couldn't be added (status 400).
+        """
+        serialized = self.get_serializer(data=request.data)
+        serialized.is_valid(raise_exception=True)
+        eregs_args = [
+            "pipeline",
+            str(serialized.validated_data["cfr_title"]),
+            str(serialized.validated_data["cfr_part"]),
+            "./testing"
+        ]
+        job = queue_eregs_job(eregs_args, timeout=60*30, result_ttl=-1)
+
+        # Paranoia--validate the values we provide:
+        job_id = job.id
+        for validator in serialized.get_fields()["job_id"].validators:
+            validator(job_id)
+        statusurl = status_url(job_id)
+        for validator in serialized.get_fields()["url"].validators:
+            validator(statusurl)
```

Comment:
I wasn't able to get that to work;  the magic provided by `rest_framework` seems to assume you're not going to want to validate anything you're providing yourself.